Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncPerformanceContext.swift

```diff
@@ -0,0 +1,232 @@
+
+    init(harness: SyncPerformanceIntTestHarness,
+         testParams: TestParams,
+         transport: FoundationInstrumentedHTTPTransport) throws
+
+    func tearDown() throws
+}
+
+extension SyncPerformanceContext {
+    func runSingleIteration(numDocs: Int,
+                            docSize: Int,
+                            testDefinition: TestDefinition,
+                            setup: SetupDefinition,
+                            teardown: TeardownDefinition) throws -> IterationResult {
+        // Perform custom setup if it exists
+        try setup(self, numDocs, docSize)
+
+        // Get starting values for disk space and time
+        let timeBefore = Date().timeIntervalSince1970
+        let networkReceivedBefore = Double(transport.bytesDownloaded)
+        let networkSentBefore = Double(transport.bytesUploaded)
+
+        var systemAttributes =
+            try FileManager.default.attributesOfFileSystem(forPath: NSHomeDirectory() as String)
+        let freeSpaceBefore =
+            (systemAttributes[FileAttributeKey.systemFreeSize] as? NSNumber)?.doubleValue ?? 0.0
+
+        // Start repeating task
+        let timeInterval = Double(testParams.dataProbeGranularityMs) / 1000.0
+        let metricsCollector = MetricsCollector(timeInterval: timeInterval)
+        metricsCollector.resume()
+
+        // Run the desired function
+        try testDefinition(self, numDocs, docSize)
+
+        // Get finish time
+        let timeMs = Double(Date().timeIntervalSince1970 - timeBefore) * 1000
+
+        // Get the point-in-time measurements
+        let pointInTimeMetrics = metricsCollector.suspend()
+
+        // Append remaining values
+        let networkReceived = Double(transport.bytesDownloaded) - networkReceivedBefore
+        let networkSent = Double(transport.bytesUploaded) - networkSentBefore
+
+        systemAttributes =
+            try FileManager.default.attributesOfFileSystem(forPath: NSHomeDirectory() as String)
+        let freeSpaceAfter =
+            (systemAttributes[FileAttributeKey.systemFreeSize] as? NSNumber)?.doubleValue ?? 0.0
+        let diskUsed = freeSpaceBefore - freeSpaceAfter
+
+        // Perform custom taredown if specified
+        try teardown(self, numDocs, docSize)
+
+        return IterationResult(time: timeMs,
+                               networkSentBytes: networkSent,
+                               networkReceivedBytes: networkReceived,
+                               cpu: pointInTimeMetrics.cpuData,
+                               memory: pointInTimeMetrics.memoryData,
+                               disk: diskUsed,
+                               threads: pointInTimeMetrics.threadData)
+    }
+
+    func waitForAllStreamsOpen() {
+        while !coll.sync.proxy.dataSynchronizer.allStreamsAreOpen {
+            print("waiting for all streams to open before doing sync pass")
+            sleep(1)
+        }
+    }
+
+    func streamAndSync() throws {
```

Comment:
[note] In the actual performance tests, we won't want to call this, because it wouldn't be an accurate simulation of real behavior. It's fine for your placeholder tests though.